include/iarc7_vision/RoombaEstimator.hpp

```diff
@@ -0,0 +1,107 @@
+#ifndef _IARC_VISION_ROOMBA_ESTIMATOR_HPP_
+#define _IARC_VISION_ROOMBA_ESTIMATOR_HPP_
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/core.hpp>
+
+#include <dynamic_reconfigure/server.h>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <ros/ros.h>
+#include <ros_utils/SafeTransformWrapper.hpp>
+#include <tf2_ros/transform_listener.h>
+
+#include "iarc7_vision/RoombaBlobDetector.hpp"
+#include "iarc7_vision/RoombaEstimatorConfig.h"
+#include "iarc7_vision/RoombaEstimatorSettings.hpp"
+
+#include <sensor_msgs/CameraInfo.h>
+#include <iarc7_msgs/RoombaDetection.h>
+#include <geometry_msgs/Point.h>
+#include <geometry_msgs/Polygon.h>
+#include <geometry_msgs/Vector3.h>
+#include <geometry_msgs/Vector3Stamped.h>
+
+namespace iarc7_vision
+{
+
+/// Gets roomba positions from bottom camera images, converts to global
+/// positions using tf, and publishes to /detected_roombas
+class RoombaEstimator {
+    public:
+        RoombaEstimator();
+
+        /// Processes current frame and publishes detections
+        ///
+        /// @param[in]  image  Current frame to process (in rgb8)
+        /// @param[in]  time   Timestamp of current frame
+        void update(const cv::cuda::GpuMat& image, const ros::Time& time);
+    private:
+
+        /// Converts a pixel in an image to a ray from the camera center
+        ///
+        /// @param[in]  px   X location of the pixel
+        /// @param[in]  py   Y location of the pixel
+        /// @param[in]  pw   Width of the image
```

Comment:
Width in pixels or in the frame of the real world? I only care about you specifying this for one parameter, it becomes self evident for the second